app/src/main/java/comp3350/mealbuddy/business/GoalTracker.java

```diff
@@ -15,6 +15,10 @@
     * takes a day and returns a list of passed goals
      */
     public static List<Goal> getPassedGoals(Calculator calc, List<Goal> goals){
+        if (calc == null || goals == null) {
+            return new ArrayList<>();
+        }
```

Comment:
this hides our error, we don't want this. we should get an exception if they're null

Reply:
true but then its less fault tolerant and we have to deal with exceptions when using it.  But it is in error state if passed null so Idk which is better. I'm fine with either

Reply:
How about we just return null?

Reply:
I think I'd rather return an empty list than null.  I say either empty list or exception

Reply:
I say exception as well